### DISCUSSION TITLE
ABB and other meters give power = 0

### DISCUSSION THREAD

**Discussion by dingo36** [dingo35/SmartEVSE-3.5#111] [General]
People who reported issues on power measurement of EV kWh meters (among others ABB), could you please test this version:

[d146e5c001.zip](https://github.com/user-attachments/files/16389720/d146e5c001.zip)

It reduces the number of reads to Eastron, ABB and Finder kWh meters; I currently have no testing capabilities for this change.

**Comment by altrnate32** [comment]
With my ABB meter it worked normally and broke somewhere the past month. Power would show 0 constantly, while other values (like total kWh) remained functioning.
With this firmware it is not 0 anymore, but also not correct. I see all kinds of values, both positive and negative (for example 17.0, -15.4, 17.6, -14.9 etc.) while actual charging speed is 11kW. It seems to alterate between positive and negative, with sometimes 2 positives or 2 negatives in a row.

**Comment by dingo35** [comment]
Thx for your report, I'll be back!

**Comment by dingo36** [comment]
This version should do better; it reads less from the meters so no performance problems on the modbus:

[97fc58d916f066.zip](https://github.com/user-attachments/files/16428922/97fc58d916f066.zip)

**Reply by bbbrumbrum** [reply]
Seems to work correct now. Thanks!

**Comment by altrnate32** [comment]
Just updated (during a charging session @ 19:49) has been correct since then.
![Screenshot 2024-07-30 195757](https://github.com/user-attachments/assets/5ea45161-f169-4bea-98bc-7c6a95008a74)

**Comment by dingo35** [comment]
Thx for your help!
